src/contracts/services/derivativeScrt.ts

```diff
@@ -0,0 +1,273 @@
+import { getActiveQueryClient$ } from '~/client';
+import {
+  switchMap,
+  first,
+  map,
+  lastValueFrom,
+} from 'rxjs';
+import { sendSecretClientContractQuery$ } from '~/client/services/clientServices';
+import { convertCoinFromUDenom } from '~/lib/utils';
+import {
+  msgQueryScrtDerivativeFees,
+  msgQueryScrtDerivativeStakingInfo,
+} from '~/contracts/definitions/derivativeScrt';
+import {
+  DerivativeScrtFeeInfoResponse,
+  DerivativeScrtStakingInfoResponse,
+} from '~/types/contracts/derivativeScrt/response';
+import {
+  DerivativeScrtFeeInfo,
+  DerivativeScrtStakingInfo,
+  DerivativeScrtInfo,
+} from '~/types/contracts/derivativeScrt/model';
+import {
+  BatchQueryParsedResponse,
+  BatchQueryParsedResponseItem,
+} from '~/types/contracts/batchQuery/model';
+import { batchQuery$ } from './batchQuery';
+
+// Contract returns price as a rate of stkd-SCRT/SCRT with 6 decimals
+const DERIVATE_PRICE_DECIMALS = 6;
+
+/**
+ * parses the response for the stkd-scrt contract info
+ */
+function parseDerivativeScrtStakingInfo(
+  data: DerivativeScrtStakingInfoResponse,
+): DerivativeScrtStakingInfo {
+  const { staking_info: stakingInfo } = data;
+  const {
+    price: exchangeRate,
+    rewards: communityRewards,
+    total_derivative_token_supply: supply,
+    unbond_amount_of_next_batch: nextUnboundAmount,
+    next_unbonding_batch_time: nextUnbondingBatchEstimatedTime,
+    validators: validatorsResp,
+  } = stakingInfo;
+
+  const validators = validatorsResp.map((nextValidator) => ({
+    validatorAddress: nextValidator.validator,
+    weight: nextValidator.weight,
+  }));
+
+  return {
+    validators,
+    supply: convertCoinFromUDenom(supply, DERIVATE_PRICE_DECIMALS).toString(),
+    exchangeRate: convertCoinFromUDenom(exchangeRate, DERIVATE_PRICE_DECIMALS).toNumber(),
+    communityRewards: convertCoinFromUDenom(communityRewards, DERIVATE_PRICE_DECIMALS).toString(),
+    nextUnboundAmount: convertCoinFromUDenom(nextUnboundAmount, DERIVATE_PRICE_DECIMALS).toString(),
+    // Seconds to Miliseconds
+    nextUnbondingBatchEstimatedTime: nextUnbondingBatchEstimatedTime * 1000,
+  };
+}
+
+/**
+ * query the staking info
+ */
+const queryDerivativeScrtStakingInfo$ = ({
+  contractAddress,
+  codeHash,
+  lcdEndpoint,
+  chainId,
+}: {
+  contractAddress: string,
+  codeHash?: string,
+  lcdEndpoint?: string,
+  chainId?: string,
+}) => getActiveQueryClient$(lcdEndpoint, chainId).pipe(
+  switchMap(({ client }: {client:any}) => sendSecretClientContractQuery$({
+    queryMsg: msgQueryScrtDerivativeStakingInfo(Math.round(new Date().getTime() / 1000)),
+    client,
+    contractAddress,
+    codeHash,
+  })),
+  map((response: any) => parseDerivativeScrtStakingInfo(
+    response as DerivativeScrtStakingInfoResponse,
+  )),
+  first(),
+);
+
+/**
+ * query the staking info
+ */
+async function queryDerivativeScrtStakingInfo({
```

Comment:
I learn towards completely removing the individual query functions and only using the batch. I cannot think of a use case where users will want data separately. This will also reduce confusion on the different between stakingInfo and allStakingInfo